src/webgpu/web_platform/canvas/getCurrentTexture.spec.ts

```diff
@@ -260,3 +265,80 @@ g.test('resize')
     currentTexture = ctx.getCurrentTexture();
     t.expect(prevTexture === currentTexture);
   });
+
+g.test('expiry')
+  .desc(
+    `
+Test automatic WebGPU canvas texture expiry on all canvas types with the following requirements:
+- getCurrentTexture returns the same texture object within the same frame, throughout:
+  - after previous frame update the rendering
+  - before current frame update the rendering
+  - in a microtask off the current frame task
+- getCurrentTexture returns a new texture object and the old texture object becomes invalid as soon as possible after HTML update the rendering.
+  `
+  )
+  .params(u =>
+    u //
+      .combine('canvasType', kAllCanvasTypes)
+      .combine('prevFrameCallsite', ['requestPostAnimationFrame', 'requestAnimationFrame'] as const)
+  )
+  .fn(t => {
+    const { canvasType, prevFrameCallsite } = t.params;
+    const ctx = t.initCanvasContext(t.params.canvasType);
+    // Create a bindGroupLayout to test invalid texture view usage later.
+    const bgl = t.device.createBindGroupLayout({
+      entries: [
+        {
+          binding: 0,
+          visibility: GPUShaderStage.COMPUTE,
+          texture: {},
+        },
+      ],
+    });
+
+    // The fn is called immediately after previous frame updating the rendering.
+    // Polyfill by calling the callback by setTimeout, in the requestAnimationFrame callback (for onscreen canvas)
+    // or after transferToImageBitmap (for offscreen canvas).
+    function requestPostAnimationFrame(fn: () => void) {
+      t.requestNewFrameOnCanvasType(canvasType, ctx, () => {
+        timeout(fn);
+      });
+    }
+
+    function checkGetCurrentTexture() {
+      // Call getCurrentTexture on previous frame.
+      const prevTexture = ctx.getCurrentTexture();
+
+      // Call getCurrentTexture immediately after the frame, the texture object should stay the same.
+      queueMicrotask(() => {
+        t.expect(prevTexture === ctx.getCurrentTexture());
+      });
+
+      // Call getCurrentTexture immediately after this frame updating the rendering.
+      // It should return a new texture object.
+      timeout(() => {
+        t.expect(prevTexture !== ctx.getCurrentTexture());
```

Comment:
I *think* we also should have a version that does *not* call ctx.getCurrentTexture() again.

That is, `prevTexture` should be expired and invalid to use, even if you don't call getCurrentTexture() again.

[queue an automatic expiry task](https://gpuweb.github.io/gpuweb/#abstract-opdef-queue-an-automatic-expiry-task) with device device and the following steps:
   - [commit the canvas texture](https://gpuweb.github.io/gpuweb/#abstract-opdef-commit-the-canvas-texture) for this.
     which says:
      - Update the rendering of context.[canvas](https://gpuweb.github.io/gpuweb/#dom-gpucanvascontext-canvas) with [a copy of the image contents](https://gpuweb.github.io/gpuweb/#abstract-opdef-get-a-copy-of-the-image-contents-of-a-context) of context.
      - If context.[[[currentTexture]]](https://gpuweb.github.io/gpuweb/#dom-gpucanvascontext-currenttexture-slot) is not null:
           - Call context.[[[currentTexture]]](https://gpuweb.github.io/gpuweb/#dom-gpucanvascontext-currenttexture-slot).[destroy()](https://gpuweb.github.io/gpuweb/#dom-gputexture-destroy) (without destroying context.[[[drawingBuffer]]](https://gpuweb.github.io/gpuweb/#dom-gpucanvascontext-drawingbuffer-slot)) to terminate write access to the image.
           - Set context.[[[currentTexture]]](https://gpuweb.github.io/gpuweb/#dom-gpucanvascontext-currenttexture-slot) to null.

Reply:
Added. Still seems to pass with Chrome